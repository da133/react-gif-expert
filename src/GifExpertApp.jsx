import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [ categories, setCategories ] = useState([ 'Greys Anatomy' ]);

    const onAddCategory = (newCategory) => {
        if( categories.includes(newCategory) ) return;

        setCategories([ newCategory, ...categories ]);
        //setCategories( cat => [...cat, 'The Office']);
    }

    return (
        <>

            {/* titulo */}
            <h1>GifExpertApp</h1>

            {/* Input */}
            <AddCategory 
                //setCategories={ setCategories }
                onNewCategory={ (value) => onAddCategory(value)}
            />

            { 
                categories.map( category => 
                    <GifGrid 
                        key={ category } 
                        category={ category }
                    />
                )
            }
       
        </>
    );
};