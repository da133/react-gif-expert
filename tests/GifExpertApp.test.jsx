import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('Puruebas en <GifExpertApp />', () => {

    test('should match initial snapshot', () => {

        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
        
    });

    test('should have initial value Greys Anatomy', () => {

        render(<GifExpertApp />);
        expect( screen.getByText('Greys Anatomy') ).toBeTruthy();
        expect( screen.getAllByText('Greys Anatomy').length ).toBe(1);

    });

});