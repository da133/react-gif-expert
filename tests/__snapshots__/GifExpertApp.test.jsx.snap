// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puruebas en <GifExpertApp /> should match initial snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Greys Anatomy
  </h3>
  <button
    class="btn-delete"
  >
    Delete
  </button>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
